botfront/imports/lib/utils.js

```diff
@@ -228,26 +223,30 @@ if (Meteor.isServer) {
             }
         },
 
-        async 'deploy.model'(projectId, target) {
+        async 'deploy.model'(projectId, target, isTest) {
             checkIfCan('nlu-data:x', projectId);
             check(target, String);
             check(projectId, String);
-            const trainedModelPath = path.join(getProjectModelLocalPath(projectId));
-            const modelFile = fs.readFileSync(trainedModelPath);
-            const { namespace } = await Projects.findOne({ _id: projectId }, { fields: { namespace: 1 } });
+            await Meteor.callWithPromise('rasa.train', projectId, 'development')
+            await Meteor.callWithPromise('commitAndPushToRemote', projectId, 'chore: deployment checkpoint')
+            const result = await runTestCases(projectId)
+            if (result.failing !== 0) {
+                throw new Meteor.Error('500', `${result.failing} test${result.failing === 1 ? '' : 's'} failed during the pre-deployment test run`)
+            }
+            const { namespace, gitSettings } = await Projects.findOne({ _id: projectId }, { fields: { namespace: 1, gitSettings: 1 } });
             const data = {
-                data: Buffer.from(modelFile).toString('base64'), // convert raw data to base64String so axios can handle it
                 projectId,
                 namespace,
                 environment: target,
-                mimeType: 'application/x-tar',
+                gitString: gitSettings?.gitString,
             };
             Meteor.call('credentials.appendWidgetSettings', projectId, target);
             const settings = GlobalSettings.findOne({ _id: 'SETTINGS' }, { fields: { 'settings.private.webhooks.deploymentWebhook': 1 } });
             const deploymentWebhook = get(settings, 'settings.private.webhooks.deploymentWebhook', {});
             const { url, method } = deploymentWebhook;
             if (!url || !method) throw new Meteor.Error('400', 'No deployment webhook defined.');
-            const resp = Meteor.call('axios.requestWithJsonBody', url, method, data);
+            // calling the webhook in a test causes it to fail
+            const resp = isTest ? { status: 200 } : Meteor.call('axios.requestWithJsonBody', url, method, data)
 
             if (resp === undefined) throw new Meteor.Error('500', 'No response from the deployment webhook');
```

Comment:
What are the possible responses from the webhook? As things stand, error responses are going to go silent in Botfront.

Perhaps better: be silent anytime `resp.?status < 300`, otherwise throw with the resp message when present.

Reply:
The webhook spec should mention all possible responses.

Reply:
This has been refactored so that if it is not 200 the error from the webhook goes through